src/Form/FieldInput.js

```diff
@@ -17,6 +17,20 @@ class FieldInput extends React.Component {
     }
   }
 
+  componentDidMount() {
+    let inputElement = ReactDOM.findDOMNode(this.refs.inputElement);
+    if (inputElement != null &&
+        inputElement.type === 'text' &&
+        this.props.focused) {
+      // Don't interfere with existing transitions
```

Comment:
Good shout. `setTimeout` is definitely a hacky workaround. Using `requestAnimationFrame` would be another option, but it still won't solve the reliability problem.

A far better approach here would be to make sure that the transition is finished by binding on the `transitionend` event, and then firing off the `focus()` call. However I am reluctant to let this component do the guesswork to find which of its ancestors it should bind itself onto... 

As we are relying on the (optional) CSS framework being use to stye these components (in our case, Canvas) and not orchestrating these transitions with JS I'm not sure what options we have left. I'm all ears in case you have an 💡 !